src/storage/txn/commands/flashback_to_version_read_phase.rs

```diff
@@ -76,80 +82,86 @@ impl CommandExt for FlashbackToVersionReadPhase {
     }
 }
 
-/// FlashbackToVersion contains two phases:
-///   1. Read phase:
-///     - Scan all locks to delete them all later.
-///     - Scan all the latest writes to flashback them all later.
-///   2. Write phase:
-///     - Delete all locks we scanned at the read phase.
-///     - Write the old MVCC version writes for the keys we scanned at the read
-///       phase.
+/// The whole flashback progress contains four phases:
+///   1. Prewrite phase:
+///     - Prewrite the `self.start_key` specifically to prevent the
+///       `resolved_ts` from advancing.
+///   2. Read-and-flashback writes phase:
+///     - Scan all the latest writes and their corresponding values at
+///       `self.version`.
+///     - Write the old MVCC version writes again for all these keys with
+///       `self.commit_ts` excluding the `self.start_key` .
+///   3. Read-and-rollback locks phase:
+///     - Scan all locks.
+///     - Rollback all these locks excluding the `self.start_key` lock we write
+///       at the first phase.
```

Comment:
I notice you swap the order of FlashbackLock and FlashbackWrite. Is there a reason for that?

Is it possible that some lock exists while you are writing the new flashbacked records? It looks a bit weird... And I just start wondering what will happen if there is a lock on the `start_key` before flashback begins...

Reply:
> And I just start wondering what will happen if there is a lock on the start_key before flashback begins...

Since all the other read/write transactions will be blocked and fail during the flashback, I think this kind of lock doesn't have any side effects. 🤔

Reply:
I'm just not sure how CDC will react if a lock is rewritten with a different start ts...